prow/tests/jobs_test.go

```diff
@@ -83,6 +85,38 @@ func TestTrustedJobs(t *testing.T) {
 	}
 }
 
+func TestPrivateJobs(t *testing.T) {
+	const (
+		private           = "private"
+		checkconfigPrefix = "gcr.io/k8s-prow/checkconfig:"
+	)
+
+	errorIfNotPermitted := func(t *testing.T, name, cluster string, spec *v1.PodSpec) {
+		if strings.Contains(cluster, private) && (len(spec.Containers) != 1 || !strings.HasPrefix(spec.Containers[0].Image, "gcr.io/k8s-prow/checkconfig:")) {
```

Comment:
I don't think this is sufficiently secure. This still allows anyone to create prowjobs for private clusters, they just have to use the checkconfig image. For example someone could create a checkconfig job and make the entrypoint an arbitrary `bash -c` command.
I think it easier and safer to only allow jobs for private clusters to be configured in a specific file/directory that we lock down with OWNERS files.

Reply:
I see, makes sense. I'll create the dir

Reply:
done